src/check_pack.c

```diff
@@ -38,16 +38,36 @@
 #endif
 
 /* Maximum size for one message in the message stream. */
-#define CK_MAX_MSG_SIZE 8192
+static int ck_max_msg_size = 0;
+#ifndef DEFAULT_MAX_MSG_SIZE
+#define DEFAULT_MAX_MSG_SIZE 4096
+#endif
 /* This is used to implement a sliding window on the receiving
  * side. When sending messages, we assure that no single message
- * is bigger than this (actually we check against CK_MAX_MSG_SIZE/2).
+ * is bigger than this.
  * The usual size for a message is less than 80 bytes.
  * All this is done instead of the previous approach to allocate (actually
  * continuously reallocate) one big chunk for the whole message stream.
  * Problems were seen in the wild with up to 4 GB reallocations.
  */
 
+void check_set_max_msg_size(int max_msg_size) {
+    if (ck_max_msg_size <= 0) {
+        char *env = getenv("CK_MAX_MSG_SIZE");
+        if (env)
+            ck_max_msg_size = atoi(env);
+        if (ck_max_msg_size <= 0)
+            ck_max_msg_size = max_msg_size;
+        if (ck_max_msg_size <= 0)
+            ck_max_msg_size = DEFAULT_MAX_MSG_SIZE;
+    }
+}
+
+static int get_max_msg_size(void) {
+    if (ck_max_msg_size <= 0)
```

Comment:
Why is this necessary? Should not the setter verify that the value is sane?

Is there a reason that ck_max_msg_size must be signed? Is it valid for it to be negative? Could it be unsigned?

Reply:
Changed type to `size_t`, which is probably the "right" type for a size.